src/Typechecker.ml

```diff
@@ -2,114 +2,121 @@ open Base
 open Result.Let_syntax
 open Ast
 open Ast.Expr
+open Location
 
 type error = string
 
 let rec check_open delta gamma expr typ =
-  match expr with
+  match expr.data with
   | Unit ->
       Result.ok_if_true
-        ([%equal: Type.t] typ Type.Unit)
-        ~error:"Expected unit type"
+        ([%equal: Type.t] typ (mkLocation Type.Unit expr.loc))
```

Comment:
```suggestion
        ([%equal: Type.t] typ (located Type.Unit))
```
`located` should use an optional argument `loc` with the default value `NoSource`.
Which means equality on types must ignore the location information.

Reply:
Done